client/header/activity-panel/unread-indicators.js

```diff
@@ -21,7 +21,8 @@ export function getUnreadNotes( select ) {
 		! Boolean( getNotesError( notesQuery ) ) &&
 		! isGetNotesRequesting( notesQuery ) &&
 		latestNote[ 0 ] &&
-		new Date( latestNote[ 0 ].date_created_gmt ).getTime() > userData.activity_panel_inbox_last_read
```

Comment:
I'm personally not a big fan of long, complex boolean checks. Would it be possible to simplify some of these into more descriptive intermediate variables at least? I think this return statement would be pretty confusing for a developer new to this code.